packages/core/stories/button/button.stories.tsx

```diff
@@ -168,3 +172,111 @@ export const FullWidth: StoryFn<typeof Button> = () => {
     </StackLayout>
   );
 };
+
+const LoadingButtonGrid = ({
+  primaryButtonLabel,
+  secondaryButtonLabel,
+  ctaButtonLabel,
+  loadingText,
+  showLoadingText,
+  withRightArrowIcon,
+}: {
+  primaryButtonLabel: string;
+  secondaryButtonLabel: string;
+  ctaButtonLabel: string;
+  loadingText: string;
+  showLoadingText?: boolean;
+  withRightArrowIcon?: boolean;
+}) => {
+  const [primaryLoadingState, setPrimaryLoadingState] = useState(false);
+  const [secondaryLoadingState, setSecondaryLoadingState] = useState(false);
+  const [ctaLoadingState, setCtaLoadingState] = useState(false);
+
+  const handlePrimaryClick = () => {
+    setPrimaryLoadingState(true);
+    setTimeout(() => {
+      setPrimaryLoadingState(false);
+    }, 3000);
+  };
+  const handleSecondaryClick = () => {
+    setSecondaryLoadingState(true);
+    setTimeout(() => {
+      setSecondaryLoadingState(false);
+    }, 3000);
+  };
+  const handleCtaClick = () => {
+    setCtaLoadingState(true);
+    setTimeout(() => {
+      setCtaLoadingState(false);
+    }, 3000);
+  };
+
+  return (
+    <div
+      style={{
+        display: "grid",
+        gridTemplateColumns: "auto auto auto",
+        gridTemplateRows: "auto",
+        gridGap: 10,
+      }}
+    >
+      <Button
+        variant="cta"
+        showLoadingText={showLoadingText}
+        loadingText={loadingText}
+        isLoading={ctaLoadingState}
+        onClick={handleCtaClick}
+      >
+        <SendIcon aria-hidden />
+        {ctaButtonLabel}
+      </Button>
+      <Button
+        variant="primary"
+        showLoadingText={showLoadingText}
+        loadingText={loadingText}
+        isLoading={primaryLoadingState}
+        onClick={handlePrimaryClick}
+      >
+        {" "}
+        {!withRightArrowIcon && <SyncIcon aria-hidden />}
+        {primaryButtonLabel}
+        {withRightArrowIcon && <ChevronRightIcon aria-hidden />}
+      </Button>
+      <Button
+        variant="secondary"
+        showLoadingText={showLoadingText}
+        loadingText={loadingText}
+        isLoading={secondaryLoadingState}
+        onClick={handleSecondaryClick}
+      >
+        {!withRightArrowIcon && <RefreshIcon aria-hidden />}
+        {secondaryButtonLabel}
+        {withRightArrowIcon && <ChevronRightIcon aria-hidden />}
+      </Button>
+    </div>
+  );
+};
+
+export const LoadingButtons: StoryFn<typeof Button> = () => {
+  return (
+    <LoadingButtonGrid
```

Comment:
I'm not a big fan of these examples. This looks like it's similar to a template but it's not using the storybook method

Reply:
I'm also not a fan of changing labels via props either

Reply:
done